magenta/music/musicxml_parser.py

```diff
@@ -544,6 +545,31 @@ def _parse_forward(self, xml_forward):
                * self.state.seconds_per_quarter)
     self.state.time_position += seconds
 
+    # Issue #674 - If the <forward> element is in a measure without
+    # any <note> elements, treat it as if it were a whole measure
+    # rest by inserting a rest of that duration
+    note_count = len(self.xml_measure.findall('./note'))
```

Comment:
I would put this section of code before any parsing of the measure happens. That way, you won't have to duplicate the code for adding the duration of the note to self.duration because it will happen as a part of normal note parsing. In order to make the timing add up, you'll probably also need to remove the <forward> element before parsing happens.